core/src/main/java/io/aiven/klaw/service/UtilControllerService.java

```diff
@@ -702,24 +706,39 @@ private Map<String, String> buildSSOProviderDetails() {
     return ssoProviders;
   }
 
-  public void resetCache(
-      int tenantId, String entityType, String entityValue, String operationType) {
+  public ApiResponse resetCache(ResetEntityCache resetEntityCache) {
+    String entityType = resetEntityCache.getEntityType();
     KwMetadataUpdates kwMetadataUpdates =
         KwMetadataUpdates.builder()
-            .tenantId(tenantId)
-            .entityType(entityType)
-            .entityValue(entityValue)
-            .operationType(operationType)
+            .tenantId(resetEntityCache.getTenantId())
+            .entityType(resetEntityCache.getEntityType())
+            .entityValue(resetEntityCache.getEntityValue())
+            .operationType(resetEntityCache.getOperationType())
             .createdTime(new Timestamp(System.currentTimeMillis()))
             .build();
+    if (getUserName().equals(ANONYMOUS_USER)) {
+      // continue with the request, ANONYMOUS_USER is generated by spring to access protected
+      // resources
+    } else if (entityType.equals(EntityType.USERS.name())
```

Comment:
Can they create their own type of entitytype like reduced_user etc?
I'm wondering would it be better to just check if it is superadmin or anon user and in any other case return unauthorized?

Reply:
- For USERS entities, we also reset cache in inmemoryauthdetails. Even if a user is added to inmemory, without a db entry, user is not authenticated. But if someone manages to run with delete operation type, that existing user cannot login. That's the risk.
- no user can call this api call directly, as we have form login
- for username field, we cannot hardcode to superadmin, as the username can be different. hence checking for relevant permission.